src/Form/Type/FormatterType.php

```diff
@@ -189,7 +191,6 @@ public function configureOptions(OptionsResolver $resolver): void
             'ckeditor_plugins' => [],
             'ckeditor_templates' => [],
             'format_field_options' => $formatFieldOptions,
-            'source_field' => null,
```

Comment:
null is a value, so, the required fields do not explain that you need to set a value.

There is a lot to be improved on this form types.